.github/workflows/benchmarking-files/regressions-util.py

```diff
@@ -0,0 +1,495 @@
+        c = colours[kind] if scale is None else colours[kind][scale]
+        return f'![#{c}](https://via.placeholder.com/15/{c}/000000?text=+)'
+
+    # Metrics, whether positive is better and the tolerances between each
+    # Neutral, kind of good/bad, very good/bad etc...
+    metric_tolerances = {
+        'acc': { 'positive_is_better': True ,'tol': [0.001, 0.01, 0.2] },
+        'auc':{ 'positive_is_better': True ,'tol': [0.001, 0.01, 0.2] },
+        'balacc': { 'positive_is_better': True ,'tol': [0.001, 0.01, 0.2] },
+        'logloss': { 'positive_is_better': False ,'tol': [0.009, 0.01, 0.2] },
+        'mae': { 'positive_is_better': False ,'tol': [0.001, 0.01, 0.2] },
+        'r2': { 'positive_is_better': True ,'tol': [0.001, 0.01, 0.2] },
+        'rmse': { 'positive_is_better': False ,'tol': [0.001, 0.01, 0.2] },
+    }
+    def is_good(score, metric):
+        return (
+            score > 0 and metric_tolerances[metric]['positive_is_better']
+            or score < 0 and not metric_tolerances[metric]['positive_is_better']
+        )
+
+    def is_neutral(diff, baseline_score, metric):
+        tolerance = metric_tolerances[metric]['tol'][0]
+        if baseline_score == 0:
+            baseline_score = 1e-10
+        prc_diff = diff / baseline_score
+        return prc_diff <= tolerance
+
+    def tolerance_colour(baseline_value, comparison_value, metric):
+        if np.isnan(baseline_value) or np.isnan(comparison_value):
+            return colour('NaN')
+
+        if baseline_value == 0:
+            baseline_value = 1e-10
+
+        prc_diff = comparison_value / baseline_value
+
+        tolerances = metric_tolerances[metric]['tol']
+        if metric_tolerances[metric]['positive_is_better']:
+            diff_color = 'Better' if prc_diff > 0 else 'Worse'
+        else:
+            diff_color = 'Better' if prc_diff < 0 else 'Worse'
+
+        if abs(prc_diff) < tolerances[0]:
+            return colour(diff_color, 0)
+        elif abs(prc_diff) < tolerances[1]:
+            return colour(diff_color, 1)
+        elif abs(prc_diff) < tolerances[2]:
+            return colour(diff_color, 2)
+        else:
+            return colour(diff_color, 3)
+
+
+    legend = {
+        'B': 'Baseline',
+        'T': 'Target Version',
+        '**Bold**': 'Training Metric',
+        '/': 'Missing Value',
+        '---': 'Missing Task'
+    }
+    legend.update({
+        key: colour(key)
+        for key in set(colours.keys()) - set(['Worse', 'Better', 'Good', 'Bad'])
+    })
+    # Worse and better are handled seperatly
+
+    compared = pd.read_csv(compared_means_csv, index_col='task')
+    baseline = pd.read_csv(baseline_means_csv, index_col='task')
+    targeted = pd.read_csv(targeted_means_csv, index_col='task')
+
+    # Some things to keep track of for the textual summary
+    n_performed_equally = 0
+    n_performed_better = 0
+    n_performed_worse = 0
+    n_could_not_compare = 0
+
+    headers = ['task', 'metric'] + METRICS
```

Comment:
Would it simplify the code if we would use [pd.DataFrame.to_markdown()](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_markdown.html)?

Reply:
There might be a way, but the complicated logic still applies as each table entry is of varying length, some contain colour, while certain entries are bold if they are the chosen metric etc...